### DISCUSSION TITLE
What is recommended way to set font that uses multiple subsets with next-intl?

### DISCUSSION THREAD

**Discussion by Demianeen** [amannn/next-intl#1535] [Q&A] [answered]
Great plugin! Much easier to use than react i18n)

To reduce the bundle size, next.js asks to specify a subset for a font. But the issue is that I need different subsets per locale. 

I tried to set it inside the layout component where I have access to the locale, but then I get error from next.js that `Font loaders must be called and assigned to a const in the module scope`. From what I understand, it is because next.js downloads fonts from google fonts during build to not make additional request every time.

Should I just specify each subset we use or is there a better way?

**Comment by amannn** [comment, marked as answer]
That's an interesting question, didn't encounter this one yet!

I had a quick look and I think `next/font` doesn't currently support dynamically loading fonts e.g. depending on props. There are however a few workarounds mentioned here if you don't mind loading the font yourself: https://github.com/vercel/next.js/discussions/40345.
